src/index.ts

```diff
@@ -107,17 +118,35 @@ export function unionize<Record>(record: Record, config?: { value?: string; tag?
     });
   }
 
+  const evalUpd = (variant: any, cases: any): any => {
+    const k: keyof Record = variant[tagProp];
+    return k in cases
+      ? creators[k](immutableUpd(payload(variant), cases[k](payload(variant))))
+      : variant;
+  };
+
+  const update = (first: any, second?: any) =>
+    second ? evalUpd(first, second) : (variant: any) => evalUpd(variant, first);
+
   return Object.assign(
     {
       is,
       as,
       match,
+      update,
       _Record: record,
     },
     creators,
   );
 }
 
+// Should we merge objects or just replace them?
+// was unable to find a better solution to that
+const objType = Object.prototype.toString.call({});
+const isObject = (maybeObj: any) => Object.prototype.toString.call(maybeObj) === objType;
```

Comment:
Hmm, I'm not sure how reliable or efficient this sort of check is. Does it even make sense to "update" something that's a primitive type? If not, could we statically forbid using this method in that case?

Reply:
I think it does make sense to have updated for primitive types (and things like Date, Array etc).
string -> string, number->number, array->array. So the output has to match exactly the input type.

But for records (plain object) it makes sense to have partial update. The best reference that I found is react setState signature. They use Pick<K,T> type and not partial.